src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1693 @@
+    """
+    Duration predictor module.
+
+    This is a module of duration predictor described in the paper 'FastSpeech: Fast, Robust and Controllable Text to
+    Speech' https://arxiv.org/pdf/1905.09263.pdf The duration predictor predicts a duration of each frame in log domain
+    from the hidden embeddings of encoder.
+
+    Note:
+        The calculation domain of outputs is different between in `forward` and in `inference`. In `forward`, the
+        outputs are calculated in log domain but in `inference`, those are calculated in linear domain.
+
+    """
+
+    def __init__(self, config: FastSpeech2ConformerConfig):
+        super().__init__()
+
+        self.conv_layers = nn.ModuleList()
+        self.log_domain_offset = 1.0
+
+        for layer_idx in range(config.duration_predictor_layers):
+            layer = FastSpeech2ConformerDurationPredictorLayer(config, layer_idx)
+            self.conv_layers.append(layer)
+        self.linear = nn.Linear(config.duration_predictor_channels, 1)
+
+    def forward(self, encoder_hidden_states, padding_masks=None, is_inference=False):
+        """
+        Args:
+            hidden_states (`torch.Tensor` of shape `(batch_size, max_text_length, input_dim)`):
+                Batch of input sequences.
+            padding_masks (`torch.ByteTensor` of shape `(batch_size, max_text_length)`, *optional*):
+                Batch of masks indicating padded part.
+            is_inference (`bool`, *optional*): Whether or not the model is running inference.
+
+        Returns:
+            `torch.Tensor`: Batch of predicted durations in log domain `(batch_size, max_text_length)`.
+
+        """
+        # (batch_size, input_dim, max_text_length)
+        hidden_states = encoder_hidden_states.transpose(1, -1)
+        for layer in self.conv_layers:
+            hidden_states = layer(hidden_states)
+
+        # NOTE: calculate in log domain, (batch_size, max_text_length)
+        hidden_states = self.linear(hidden_states.transpose(1, -1)).squeeze(-1)
+
+        if is_inference:
+            # NOTE: calculate in linear domain
+            hidden_states = torch.clamp(torch.round(hidden_states.exp() - self.log_domain_offset), min=0).long()
+
+        if padding_masks is not None:
+            hidden_states = hidden_states.masked_fill(padding_masks, 0.0)
+
+        return hidden_states
+
+
+class FastSpeech2ConformerLengthRegulator(nn.Module):
+    """
+    Length regulator module for feed-forward Transformer.
+
+    This is a module of length regulator described in `FastSpeech: Fast, Robust and Controllable Text to Speech`
+    https://arxiv.org/pdf/1905.09263.pdf. The length regulator expands char or phoneme-level embedding features to
+    frame-level by repeating each feature based on the corresponding predicted durations.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, encoded_embeddings, duration_labels, speaking_speed=1.0):
```

Comment:
Yea it's actually not an arg in the model's forward method, the value comes from the config, but I believe we still want to pass it to the length regulator's forward so it can still be configured easily after the model is instantiated. Updated it to be updated by the `config.speaking_speed` instead of the `model.speaking_speed`.

Are you also saying you'd prefer to send the whole config on the forward call?

(from the model forward)
```py
hidden_states = self.length_regulator(hidden_states, duration_predictions, self.config.speaking_speed)
```

Reply:
I was thinking that we just set it in the `__init__` of the length regulator directly from the config:
```python
    def __init__(self, config):
        super().__init__()
        self.speaking_speed = config.speaking_speed
```

I'm not sure how much the `speaking_speed` attribute would actually be changed in practice, which is why I think it's fine just to set once in the config and not worry about updating it.

If we disagree with this, and want to let the user configure the speaking speed, then let's keep the `speaking_speed` argument to the length regulator, and add a line in `FastSpeech2ConformerModel`:
```python
speaking_speed = speaking_speed if speaking_speed is not None else self.config.speaking_speed
```

cc @Vaibhavs10 @ylacombe